hw2/hw2.sql

```diff
@@ -0,0 +1,57 @@
+use baseball;
+# Question 1 -  Annual
+
+#drop table before creating a new
+
+drop table annual_average;
+
+#create a new table
+
+create table annual_average
+select DISTINCT a.batter,extract (year from g.local_date) as year ,(sum(a.Hit)/NULLIF(sum(a.atBat),0))*100 as batting_average
+from
+batter_counts as A
+JOIN game g on A.game_id = g.game_id
+group by a.batter, year;
+
+
+
+#Question 2 - Historical
+
+drop table historical_average;
+
+#create a new table
+
+create table historical_average
+select DISTINCT a.batter,(sum(a.Hit)/NULLIF(sum(a.atBat),0))*100 as batting_average
+from
+batter_counts as A
+JOIN game g on A.game_id = g.game_id
+group by a.batter;
+
+#Question 3
+
+#Q3 - Probably Correct
+
+#Step 1 - Create a temp table with all required columns
+
+#drop table before creation
+
+create or replace table rolling_average
+with rolling_average_intermediate as
+(SELECT a.batter, b.local_date, A.hit,A.atBat
```

Comment:
I think it might be better not always use the same aliases for your tables. It gets confusing when all the tables are switched between A, B, a, b.

Reply:
totally makes sense.